pkg/providers/tinkerbell/comparator.go

```diff
@@ -0,0 +1,51 @@
+package tinkerbell
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+// VersionComparator enables kubernetes version retrieval.
+type VersionComparator interface {
+	GetVersion() string
+}
+
+// ReplicaComparator enables replica number retrieval.
+type ReplicaComparator interface {
+	GetReplicas() int
+}
+
+// KubeadmControlPlaneComparator wraps the KubeadmControlPlane to enable Comparator functions.
+type KubeadmControlPlaneComparator struct {
+	*controlplanev1.KubeadmControlPlane
+}
+
+// GetVersion retrieves KubeadmControlPlane replicas.
+func (k *KubeadmControlPlaneComparator) GetVersion() string {
```

Comment:
prefer accessors without the `Get`
```suggestion
func (k *KubeadmControlPlaneComparator) Version() string {
```

https://go.dev/doc/effective_go#Getters